### PR TITLE
remove news.featured

### PR DESCRIPTION
**Pull request by Pomax**
remove featured flag use across the codebase, as prerequisite to removing the field itself from news.models in followup.

Part 1 of https://github.com/mozilla/foundation.mozilla.org/issues/721
Unblocks https://github.com/mozilla/foundation.mozilla.org/pull/2861

### CODE REVIEW

**Comment by alanmoo** [comment]
That's an odd Travis failure

**Comment by alanmoo** [comment]
Also, is this dramatic Percy change the normal "random data seed generation" thing? If so, this is getting to be...not useful.

**Comment by Pomax** [comment]
Errrr... I'll hard-coding a PNI product, because we shouldn't guess which one the faker invents for us >_>

**Comment by alanmoo** [comment]
Ah, so that's still to be done on this PR?

File: cypress/integration/endpoint-tests.js [hunk 1]
```diff
@@ -104,7 +104,7 @@ describe(`Visual regression testing for foundation.mozilla.org`, () => {
   });
 
   it(`PNI product page`, function() {
-    cy.visit(`/en/privacynotincluded/products/home-great/`);
+    cy.visit(`/en/privacynotincluded/products/near-reality/`);
```
Comment (alanmoo) [inline cypress/integration/endpoint-tests.js]: I see you changed this after your comment, but I assume there's another commit to come that manually adds this product to the fake database?

**Comment by Pomax** [comment]
blocked until the reseed PR (#2897) lands.

File: cypress/integration/endpoint-tests.js (hunk 1 above)
Comment (Pomax) [reply cypress/integration/endpoint-tests.js]: yeah, handled in #2897 after which I'll rebase this.

**Pull request merged**